platform/kendryte/k210/platform.c

```diff
@@ -67,16 +67,26 @@ static u32 k210_get_clk_freq(void)
 
 	return pll0_freq / div;
 }
+volatile static u64 MMUTable[4096 / sizeof(u64)] __attribute__((aligned(4 * 1024))) = {0};
 
 static int k210_final_init(bool cold_boot)
 {
 	void *fdt;
 
+	u64 ms = csr_read(mstatus);
+	ms &= ~(0xf << 24);
+	ms |= 9 << 24;
+	csr_write(mstatus,ms);
+	MMUTable[0] = 0x0000002fULL;
+	MMUTable[1] = 0x1000002fULL;
+	MMUTable[2] = 0x2000002fULL;
+	MMUTable[3] = 0x3000002fULL;
+	csr_write(sptbr,((u64)MMUTable) >> 12);
+	
```

Comment:
Does this mean that you are unconditionally enabling the MMU here if we jump to S-mode for the payload ? Where is the control for choosing which mode we will jump to ?

Reply:
Yes,the mmu is unconditionally enabled now and there's no any other control,maybe i can add a configuration item?

Reply:
But actually that,for K210,if user doesn't enable MMU,they must use the range of 0x00000000~0xffffffff which contains all available physical addresses,the default map pattern is one-to-one mapping for 0x00000000~0xffffffff, so maybe it doesn't cause any problem I think.